Presentation.Web/app/services/select2LoadingService.ts

```diff
@@ -35,6 +36,14 @@
             };
         }
 
+        select2LocalDataFormattedIndentation(dataFn: () => Models.ViewModel.Generic.Select2OptionViewModelWithIndentation<any>[], formatResults: (input) => string, allowClear= true) {
```

Comment:
I'm not a fan of this solution of creating a second method of "select2LocalDataFormatted", but I didn't know how to get around that for `indentation` I need to use another model, which doesn't extend the Select2OptionViewModel.

The difference between the models:
-Select2OptionViewModel: id: number
-Select2OptionViewModelWithIndentation: id: string (extends ISelect2Model)

Reply:
notice the `<>`??
right now it is hard coded to "any", but if you make it a generic type param you can add that type param to the formatting function.. in stead of adding a new "select2optionmodel...."... you just change the type in the `T` and the rest will tag along...

I think the main issue is that in stead of using the optionalObjectContext, the "select2withindent" was introduced in stead of using proper generics... If in doobt... come ask